plugins/TimingTriggerCandidateMaker.cpp

```diff
@@ -1,55 +1,144 @@
-		}
-
-
-		TriggerCandidate TriggerCandidateMakerTiming::TimeStampedDataToTriggerCandidate(const TimeStampedData& data) {
-			//Fill unused fields
-			std::vector<uint16_t> detid_list;
-			std::vector<TriggerPrimitive> primitive_list;
-			std::vector<TriggerActivity> activity_list;
-
-			uint32_t detid = 0;
-			TriggerPrimitive primitive {0,0,0,0,0,0,0,0,0,0,0};
-			TriggerActivity activity {0,0,0,0,0,0,0,0,0,0,0,0,0,primitive_list};
-
-			detid_list.push_back(detid);
-			primitive_list.push_back(primitive);
-			activity_list.push_back(activity);
-
-			TriggerCandidate candidate;
-			candidate.time_start =  time - m_map[data.signal_type].first, // time_start
-			candidate.time_end = time + m_map[data.signal_type].second, // time_end, 
-			candidate.time_candidate = int64_t(clock(now.time_since_epoch()).count()); 
-			candidate.detid = detid_list;
-			candidate.type = data.signal_type;
-			candidate.algorithm = data.counter;
-			candidate.version = 0;
-			candidate.activity_list = activity_list;
-
-			return candidate;
-		}
-
-		void DAQTriggerCandidateMaker::do_conf(const nlohmann::json& config /*args*/) {
-			auto params = config.get<dunedaq::trigger::timingtriggercandidatemaker::Conf>();
-		}
-
-	} //namespace trigger
-}	//namespace dunedaq
+namespace trigger {
+TimingTriggerCandidateMaker::TimingTriggerCandidateMaker(const std::string& name)
+  : DAQModule(name)
+  , thread_(std::bind(&TimingTriggerCandidateMaker::do_work, this, std::placeholders::_1))
+  , inputQueue_(nullptr)
+  , outputQueue_(nullptr)
+  , queueTimeout_(100)
+{
+
+  register_command("conf", &TimingTriggerCandidateMaker::do_conf);
+  register_command("start", &TimingTriggerCandidateMaker::do_start);
+  register_command("stop", &TimingTriggerCandidateMaker::do_stop);
+  register_command("scrap", &TimingTriggerCandidateMaker::do_scrap);
+}
+
+triggeralgs::TriggerCandidate
+TimingTriggerCandidateMaker::TimeStampedDataToTriggerCandidate(const triggeralgs::TimeStampedData& data)
+{
+  // Fill unused fields
+  std::vector<uint16_t> detid_list;
+  std::vector<triggeralgs::TriggerPrimitive> primitive_list;
+  std::vector<triggeralgs::TriggerActivity> activity_list;
+
+  uint32_t detid = data.signal_type;
+  triggeralgs::TriggerPrimitive primitive{ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+  triggeralgs::TriggerActivity activity{ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, primitive_list };
+
+  detid_list.push_back(detid);
+  primitive_list.push_back(primitive);
+  activity_list.push_back(activity);
+
+  triggeralgs::TriggerCandidate candidate;
+  auto now = std::chrono::steady_clock::now();
+
+  candidate.time_start = data.time_stamp - m_map[data.signal_type].first,  // time_start
+  candidate.time_end = data.time_stamp + m_map[data.signal_type].second, // time_end,
+  candidate.time_candidate = data.time_stamp;
+  candidate.detid = detid_list;
+  candidate.type = TriggerCandidateType::kTiming;
+  candidate.algorithm = uint32_t(internal_clock(now.time_since_epoch()).count());
```

Comment:
I don't understand this line. Why are you setting the algorithm type to the system clock time?